portal-impl/src/com/liferay/portal/upgrade/v7_0_5/UpgradeThemeId.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.upgrade.v7_0_5;
+
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.StringBundler;
+
+/**
+ * @author Michael Bowerman
+ */
+public class UpgradeThemeId extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		for (String[] themeIds : _RENAMED_THEME_IDS) {
+			String oldThemeId = themeIds[0];
+			String newThemeId = themeIds[1];
+
+			runSQL(
+				StringBundler.concat(
+					"update LayoutSet set themeId = '", newThemeId,
+					"' where themeId = '", oldThemeId, "'"));
+		}
+	}
+
+	private static final String[][] _RENAMED_THEME_IDS = {
+		new String[] {"classic", "classic_WAR_classictheme"},
+		new String[] {"admin", "admin_WAR_admintheme"}
```

Comment:
Hi @mbowerman, @SamZiemer,

The pull looks good but I have one question, which is the admin theme? where can I find it to deploy it in my tests?

Thanks.

Reply:
Hi @achaparro, the admin theme can be found in modules/apps/foundation/frontend-theme/frontend-theme-admin

Reply:
@SamZiemer does it exist in previous versions of Liferay (6.2 and below)?

Thanks.

Reply:
Hmm, I'm not sure where the admin theme is in 6.2 or below.

Reply:
Then we don't need to upgrade it, right? admin can be a common word for naming a theme in custom developments so if we didn't use it in 6.2 and below we shouldn't upgrade it. Please, correct me if I am wrong.

Thanks.

Reply:
It looks like it was called "controlpanel" in 6.2 and earlier. See https://github.com/liferay/liferay-portal-ee/commit/161b85f7823da9ab42bced1fcfb3f13a3c477ae7. So maybe we should add a third value to the array that maps "controlpanel" to "admin_WAR_admintheme"?

Reply:
We would need to upgrade it for customers who were on a version of DXP that were on the base version of DXP (without LPS-65845). Actually, I'm not even sure if it's possible to select the Admin theme for a LayoutSet. So maybe we wouldn't need to upgrade it anyway.

Reply:
Hi @mbowerman, thanks for the info. In that case I think that best option is to have just two options replacing "admin" for controlpanel.

is that ok for you?

Reply:
Sorry, I don't understand your proposal. I think we should do this:
    
```
private static final String[][] _RENAMED_THEME_IDS = {
	new String[] {"classic", "classic_WAR_classictheme"},
	new String[] {"admin", "admin_WAR_admintheme"},
	new String[] {"controlpanel", "admin_WAR_admintheme"}
};
```

That way it upgrades properly for people who are coming from 6.2 or earlier, as well as for people who are coming from the base DXP installation. Is this what you had in mind?